agimus_controller/agimus_controller/warm_start_reference.py

```diff
@@ -0,0 +1,18 @@
+from agimus_controller.warm_start_base import WarmStartBase
+from agimus_controller.trajectory import TrajectoryPoint
+import numpy as np
+
+
+class WarmStartReference(WarmStartBase):
+    def generate(self, reference_trajectory: list[TrajectoryPoint]):
```

Comment:
Ok this is wrong ;)
the `mpc.py` uses it as:
```python
x0, x_init, u_init = self._warm_start.generate(
            initial_state, reference_trajectory, self._ocp.debug_data.result
        )
```

Reply:
ah, missed this change, will update

Reply:
It is. But it is meant to convert the custom class representation into Numpy array that is tied to your robot

Reply:
yes, i.e. the `x0 = initial_state` here

Reply:
initial state is a Trajectory Point